poleno/utils/templatetags/poleno/utils.py

```diff
@@ -337,8 +337,14 @@ def change_lang(context, lang=None):
     try:
         url_parts = resolve(path)
         view_name = url_parts.view_name
+        args = url_parts.args
         kwargs = url_parts.kwargs
 
+        # Mix *args and **kwargs in call to reverse
```

Comment:
@martinmacko47 Este som rozmyslal nad tym, ci pridanie `args` dam pred `if hasattr ...` cast alebo az za. Ale ak ma `url_parts.func` attribut `change_lang` (alebo ak v buducnosti pribudne iny view s tymto attributom), tak tam to je podmienen ze `kwargs` argumenty budu spravne podla definovaneho view a `args` budu vtedy prazdne. Takze nezalezi kde sa argumenty pridaju, ci pred alebo za castou `hasattr`. 
```
@decorate(change_lang=change_lang)
@require_http_methods([u'HEAD', u'GET'])
def view(request, path):
    ...
```


Ale nebolo by odveci keby som to dal pod `else` klauzulu t.j.
```
if hasattr(url_parts.func, u'change_lang'):
    view_name, kwargs = url_parts.func.change_lang(lang, **kwargs)
else:
    # Mix *args and **kwargs in call to reverse
    for i in range(len(args)):
        name = u'_{}'.format(i)
        kwargs[name] = args[i]
```

Reply:
@viliambalaz Nestaci proste poslat do `reverse` aj `args` aj `kwargs`?

```
url = reverse(view_name, args=args, kwargs=kwargs)
```

Ak tomu spravne rozumiem, posielat sucastne `args` aj `kwargs` do `reverse` je zakazane preto, ze je zakazane mat v jednej URL naraz aj pozicne aj pomenovane parametre. To vsak znemena, ze vzdy by aspon jedno z `args` alebo `kwargs` malo byt prazdne, takze realne sa nikdy neposlu oba.

Analogicky pri `func.change_lang`. IMHO poslat tam aj `args` aj `kwargs`. Je na zodpovednosti autora viewu, aby `func.change_lang` ocakaval take parametre, ake URL smerujuca na dany view obsahuje. Cize:

```
view_name, kwargs = url_parts.func.change_lang(lang, *args, **kwargs)
```

Resp. by som `func.change_lang` zovseobecnil az tak, aby aj vracalo aj `args` aj `kwargs`. Aby URL po zmene jazyka tiez mohlo obsahovat bud pozicne parametre alebo pomenovane parametre:

```
view_name, args, kwargs = url_parts.func.change_lang(lang, *args, **kwargs)
```

S tym, ze samozrejme bude treba upravit implementacie vsetkych `func.change_lang`, ktore v kode mame. Aby vracali aj `args`. Zrejme to bude vsade prazdne pole.

Reply:
:ok: